### ISSUE TITLE
UCD (Unified Change Detection) Local Setup Guide For Testing

### ISSUE THREAD

**Issue by Avicii786** [AI4RS/UCD#3]
# UCD (Unified Change Detection) Local Setup Guide For Testing

Note: This guide is only for testing based on pre-trained weights that author is providing. You can adjust it for other dataset or model. Here I illustrated an example of SYSUCD dataset and BIT model.

This guide summarizes the steps to set up the UCD repository locally, handle environment issues (especially for new NVIDIA GPUs like the RTX 50-series), pre-process a local dataset, and run a test.

---

## 1. ⚙️ Environment Setup (Do This Once)

This ensures your `ucd` conda environment has the correct PyTorch version for your GPU and all necessary dependencies.

### 1. Clone the Repository
```bash
git clone [https://github.com/AI4RS/UCD](https://github.com/AI4RS/UCD)
cd UCD

conda create -n ucd python=3.10 -y
conda activate ucd
# you can install latest pytorch version from website
#pip install --pre torch torchvision torchaudio --index-url [https://download.pytorch.org/whl/nightly/cu121](https://download.pytorch.org/whl/nightly/cu121)
pip install -r requirements.txt
pip install --upgrade evaluate huggingface_hub transformers
pip install scikit-learn
pip install pyyaml idna requests
accelerate config #after running this command it will ask you some questions about your environment setup. Select single GPU option
```

### 2. 🗂️ Data Pre-processing (Do This Once Per Dataset)

```code
The script cannot read raw image folders. You must convert your dataset into the Hugging Face DatasetDict format first using the dataset_maker.py script.
```
#### 2.1. Organize Raw Data Folders
Ensure your dataset subfolders are named exactly imageA, imageB, and label.

```code
D:/your_dir/datasets/sysu/
├── train/
│   ├── imageA/
│   ├── imageB/
│   └── label/
├── val/
│   ├── imageA/
│   ├── imageB/
│   └── label/
└── test/
    ├── imageA/
    ├── imageB/
    └── label/
```

#### 2.2. Edit `dataset_maker.py`
Open the `dataset_maker.py` file in the UCD folder and make two changes:

Update all paths at the top of the script to point to your raw data folders (e.g., D:/your_dir/datasets/sysu/train/imageA/).

Change the very last line from dataset.push_to_hub() to dataset.save_to_disk() and give it a new folder name for the output.

Example(Bottom of the line)
```python
# dataset.push_to_hub("")  # <-- Comment this out or delete it
dataset.save_to_disk("my_processed_sysu_dataset")
print("Dataset successfully saved to disk.")
```
#### 2.3. Run the `dataset_make.py` Script
From your UCD folder (with the (ucd) environment active):
```bash
python dataset_maker.py
```
This creates a new folder (e.g., my_processed_sysu_dataset) inside your UCD directory. This new folder contains the processed .arrow files that the test script needs.

### 3. 🚀 Run Testing (The Final Command)
You do not need to use the .yml config file for just testing. Run the test by passing the arguments directly, as specified in the repository's README.

Remember to replace the paths with your specific paths.

```bash
accelerate launch test.py --model HZDR-FWGEL/UCD-SYSUCD-BIT --dataset-path "D:/your_dir/datasets/sysu/" --dataset-name SYSUCD
```

Command Breakdown:

`--model HZDR-FWGEL/UCD-SYSUCD-BIT`: Downloads and loads the correct, pre-trained BIT model weights.

`--dataset-path "..."`: Points to your processed dataset folder (the one created by dataset_maker.py).

`--dataset-name SYSUCD`: Tells the script to use the "SYSUCD" data loading logic.
